cli/pcluster/cfnconfig.py

```diff
@@ -354,6 +354,11 @@ def __init_vpc_parameters(self):
                     "VPC section [%s] used in [%s] section is not defined" % (vpc_section, self.__cluster_section)
                 )
 
+        # Check that cidr and public ips are not both set
+        cidr_value = self.__config.get(vpc_section, "compute_subnet_cidr", fallback=None)
+        public_ips = self.__config.getboolean(vpc_section,"use_public_ips", fallback=True)
+        self.__validate_resource("VPC_COHERENCE", (cidr_value, public_ips))
```

Comment:
if possible let's avoid adding more and more code to this __validate_resource function :)

Reply:
how would you do that if not adding code there?

Reply:
You can extract the code under the `elif resource_type == "VPC_COHERENCE":` branch into a separate function (something like validate_vpc_coherence) and call that in place of the huge validate function.

Reply:
Done, check if it is what you meant